.github/workflows/build-and-publish.yaml

```diff
@@ -8,7 +8,9 @@ on:
 jobs:
   build-and-publish:
     name: Java Gradle
-    uses: bakdata/ci-templates/.github/workflows/java-gradle-library.yaml@1.39.1
+    uses: bakdata/ci-templates/.github/workflows/java-gradle-library.yaml@1.40.6
+    with:
+      java-version: 17
```

Comment:
Yes. We need to use Java 17 in order to submit to Sonarqube. However, all code is compiled as Java 11 (we do this in all projects already, this is just missing because we migrated it to GitHub first)